asset-transfer-basic/chaincode-javascript/test/assetTransfer.test.js

```diff
@@ -0,0 +1,251 @@
+
+            let ret = JSON.parse(await chaincodeStub.getState('asset1'));
+            expect(ret.ID).to.equal('asset1');
+        });
+    });
+
+    describe('Test UpdateAsset', () => {
+        it('should return error on UpdateAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            try {
+                await assetTransfer.UpdateAsset(transactionContext, 'asset2', 'orange', 10, 'Me', 500);
+                assert.fail('UpdateAsset should have failed');
+            } catch (err) {
+                expect(err.message).to.equal('The asset asset2 does not exist');
+            }
+        });
+
+        it('should return success on UpdateAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            await assetTransfer.UpdateAsset(transactionContext, 'asset1', 'orange', 10, 'Me', 500);
+            let ret = JSON.parse(await chaincodeStub.getState('asset1'));
+            expect(ret.ID).to.equal('asset1');
+            expect(ret.Color).to.equal('orange');
+            expect(ret.Size).to.equal(10);
+            expect(ret.Owner).to.equal('Me');
+            expect(ret.AppraisedValue).to.equal(500);
+        });
+    });
+
+    describe('Test DeleteAsset', () => {
+        it('should return error on DeleteAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            try {
+                await assetTransfer.DeleteAsset(transactionContext, 'asset2');
+                assert.fail('DeleteAsset should have failed');
+            } catch (err) {
+                expect(err.message).to.equal('The asset asset2 does not exist');
+            }
+        });
+
+        it('should return success on DeleteAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            await assetTransfer.DeleteAsset(transactionContext, 'asset1');
+            let ret = await chaincodeStub.getState('asset1');
+            expect(ret).to.equal(undefined);
+        });
+    });
+
+    describe('Test TransferAsset', () => {
+        it('should return error on TransferAsset', async () => {
+            let assetTransfer = new AssetTransfer();
+            await assetTransfer.CreateAsset(transactionContext, 'asset1', 'blue', 5, 'Tomoko', 300);
+
+            try {
+                await assetTransfer.TransferAsset(transactionContext, 'asset2', 'Me');
+                assert.fail('DeleteAsset should have failed');
+            } catch (err) {
+                assert.pass();
```

Comment:
Compare the error message `expect(err.message).to.equal('The asset asset2 does not exist');`

Reply:
Fixed as indicated